seerep-hdf5/seerep-hdf5-fb/src/hdf5-fb-general.cpp

```diff
@@ -11,56 +11,53 @@ Hdf5FbGeneral::Hdf5FbGeneral(std::shared_ptr<HighFive::File>& file, std::shared_
 
 void Hdf5FbGeneral::writeAttributeMap(
     const std::shared_ptr<HighFive::DataSet> dataSetPtr,
-    const flatbuffers::Vector<flatbuffers::Offset<seerep::fb::UnionMapEntry>>* attributes)
+    const flatbuffers::Vector<flatbuffers::Offset<seerep::fb::UnionMapEntry>>& attributes)
 {
-  if (attributes)
+  for (auto attribute : attributes)
```

Comment:
Are you sure that this won't create a segmentation fault, when the attribute map isn't set in the FB msg?